src/ok.rs

```diff
@@ -178,6 +178,45 @@ fn collect_tests() -> Vec<Test> {
                             }),
                             sub: vec![],
                         },
+                        Test {
+                            name: "Secure Nested Paging (SEV-SNP)",
+                            gen_mask: SNP_MASK,
+                            run: Box::new(|| {
+                                let res = unsafe { x86_64::__cpuid(0x8000001f) };
+
+                                let stat = if (res.eax & 0x1 << 4) != 0 {
+                                    TestState::Pass
+                                } else {
+                                    TestState::Fail
+                                };
+
+                                TestResult {
+                                    name: "Secure Nested Paging (SEV-SNP)",
+                                    stat,
+                                    mesg: None,
+                                }
+                            }),
+                            sub: vec![Test {
+                                name: "VM Permission Levels",
+                                gen_mask: SNP_MASK,
+                                run: Box::new(|| {
+                                    let res = unsafe { x86_64::__cpuid(0x8000001f) };
+
+                                    let stat = if (res.eax & 0x1 << 5) != 0 {
+                                        TestState::Pass
+                                    } else {
+                                        TestState::Fail
+                                    };
+
+                                    TestResult {
+                                        name: "VM Permission Levels",
+                                        stat,
+                                        mesg: None,
+                                    }
+                                }),
+                                sub: vec![],
```

Comment:
Checking the number of VMPLs available would be a good sub-test to this one. This can be done with CPUID as well.